src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/KeyReadsByUUID.java

```diff
@@ -0,0 +1,29 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.UUID;
+
+/**
+ * KeyReadsByUUID takes a PCollection of reads and produces one KV per read where the key is the UUID from the read.
+ * This is necessary to use reads in GroupBy and CoGroupBy since GATKRead is cannot be deterministically encoded.
+ */
+public class KeyReadsByUUID extends PTransform<PCollection<GATKRead>, PCollection<KV<UUID, GATKRead>>> {
```

Comment:
Wait, we're adding a 16-bytes field to every read? Are we sure about this? We have tons of reads, this is going to add up.